.gitlab-ci.yml

```
@@ -3470,6 +3475,17 @@ deploy_staging_datadog_agent_windows_zip:
   script:
     - $S3_CP_CMD --recursive --exclude "*" --include "datadog-agent-7.*.zip" $OMNIBUS_PACKAGE_DIR $S3_DSD6_URI/windows/agent7/bosh/ --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
 
+deploy_cluster_agent_cloudfoundry:
+  <<: *run_only_when_triggered
+  <<: *skip_when_unwanted_on_7
+  stage: deploy7
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/deploy:$DATADOG_AGENT_BUILDERS
+  before_script:
+    - ls $OMNIBUS_PACKAGE_DIR
```

Let's either:
- download only necessary artifacts:
```suggestion
  dependencies:
    - cluster_agent_cloudfoundry-build_amd64
  before_script:
    - ls $OMNIBUS_PACKAGE_DIR
```
- add an explicit `needs` on `cluster_agent_cloudfoundry-build_amd64` (what `deploy_staging_datadog_agent_windows_binaries_zip` is doing)
```suggestion
  needs:
    - cluster_agent_cloudfoundry-build_amd64
  before_script:
    - ls $OMNIBUS_PACKAGE_DIR
```

---

Which option do you like better? I was doing the second, but then I realized the deploy would start early (and maybe the build for some other platform would fail) and undid. The first option is better in that regard.

---

I like the first option better because it's more conservative: we are assured that all our tests pass before deploying anything.

However, at some point in the future we'll need to harmonize the deploy jobs, as some are using `needs` while others are using `dependencies`.